src/west/manifest.py

```diff
@@ -537,14 +573,72 @@ def _load_project(self, pdata):
         except ValueError as ve:
             self._malformed(ve.args[0])
 
+        import_files = self._imported_files(project, import_key)
+
         # The name "manifest" cannot be used as a project name; it
         # is reserved to refer to the manifest repository itself
         # (e.g. from "west list"). Note that this has not always
         # been enforced, but it is part of the documentation.
         if project.name == 'manifest':
             self._malformed('no project can be named "manifest"')
 
-        return project
+        return project, import_files
+
+    def _imported_files(self, project, import_key):
+        ktype = type(import_key)
+
+        if not import_key or (isinstance(project, ManifestProject) and
+                              ktype == bool):
+            return []
+
+        if ktype == bool:
```

Comment:
What does `import: true` actually mean in the context of the `self` section?

Reply:
Nothing. It's ignored there.

https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/20433/docs/guides/west/manifest.html#option-1-boolean